evm/interchainTokenFactory.js

```diff
@@ -207,6 +209,46 @@ async function processCommand(config, chain, options) {
             break;
         }
 
+        case 'registerAllGatewayTokens': {
+            let { assetApi, tokenInfoApi, batchSize } = options;
+
+            assetApi = assetApi || `${config.axelar.lcd}/axelar/nexus/v1beta1/assets/`;
+            tokenInfoApi = tokenInfoApi || `${config.axelar.lcd}/axelar/evm/v1beta1/token_info/`;
+            batchSize = batchSize || 10;
+
+            validateParameters({ isString: { assetApi, tokenInfoApi }, isNumber: { batchSize } });
+
+            const { assets } = (await axios.get(`${assetApi}${chain.name}`)).data;
+            const unregisteredAssets = [];
+
+            for (const asset of assets) {
+                const salt = keccak256(hexlify(toUtf8Bytes(asset)));
+                const tokenId = await interchainTokenService.interchainTokenId(AddressZero, salt);
+                const tokenManagerAddress = await interchainTokenService.tokenManagerAddress(tokenId);
+
+                if ((await provider.getCode(tokenManagerAddress)).length === 2) {
+                    unregisteredAssets.push(asset);
+                }
+            }
+
+            for (let i = 0; i < unregisteredAssets.length; i += batchSize) {
+                const multicallData = [];
+
+                for (let j = i; j < i + batchSize && j < unregisteredAssets.length; j++) {
+                    const asset = unregisteredAssets[j];
+                    const salt = keccak256(hexlify(toUtf8Bytes(asset)));
+                    const { symbol } = (await axios.get(`${tokenInfoApi}${chain.name}?asset=${asset}`)).data.details;
+                    const { data } = await interchainTokenFactory.populateTransaction.registerGatewayToken(salt, symbol);
+                    multicallData.push(data);
+                }
+
+                const tx = await interchainTokenFactory.multicall(multicallData, { gasOptions });
```

Comment:
```suggestion
                const tx = await interchainTokenFactory.multicall(multicallData, gasOptions);
```
`getGasOptions` should return `gasOptions` as an object already